src/directConnectManager.ts

```diff
@@ -157,9 +171,18 @@ export class DirectConnectionManager {
     ResourceLoader.deregisterInstance(id);
   }
 
-  async updateConnection(spec: CustomConnectionSpec): Promise<void> {
+  async updateConnection(incomingSpec: CustomConnectionSpec): Promise<void> {
+    // at this point incoming spec has placeholder secrets... look up the associated ConnectionSpec
+    const currentSpec: ConnectionSpec | null = await getResourceManager().getDirectConnection(
```

Comment:
I can't imagine this change would actually affect much at the moment, but minor type update:
```suggestion
    const currentSpec: CustomConnectionSpec | null = await getResourceManager().getDirectConnection(
```

Reply:
causes TS errors.... Having and translating these two types is a bit of a mess actually